sktime/forecasting/base/_base.py

```diff
@@ -216,18 +173,8 @@ def predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
                 "an issue on sktime."
             )
 
-        # input check for X
-        enforce_index_type = self.get_tag("enforce_index_type")
-        X = check_series(X, enforce_index_type=enforce_index_type, var_name="X")
-
-        # convert X if needed
-        X_inner_mtype = self.get_tag("X_inner_mtype")
-        X_inner = convert_to(
-            X,
-            to_type=X_inner_mtype,
-            as_scitype="Series",  # we are dealing with series
-            store=None,
-        )
+        # input check and conversion for X
+        X_inner, _ = self._check_X_y(X=X)
```

Comment:
Would it make sense to have a separate `_check_X` method, that could be then be called inside `_check_X_y` and would avoid the somewhat awkward second return object. With the new probabilistic prediction methods there will be a number of methods that only need to check X. 
```suggestion
        X_inner = self._check_X(X)
```

Reply:
how about a simple shorthand instead, i.e., `_check_X()` is `_check_X_y()[0]`?

In terms of flow logic, that's much simpler than your proposed "splitting up" of `_check_X_y`, and easier to understand (since the code in `check_X_y` is ordered by concerns across X/y, whereas the splitting would create two functions in which some concerns are contained in `_check_X_y` and some in `check_X`).

Reply:
I find it unintuitive to see `check_X_y` in methods that do not take in `y`. So I would prefer splitting the methods, I think it should be straightforward, but I may be missing something.